File: app/src/main/java/com/acceleratorer/wuwavn/PatchDryRun.java
package com.acceleratorer.wuwavn;

import java.io.File;
import java.util.List;

final class PatchDryRun {
    final List<String> filesToAdd;
    final List<String> filesToModify;
    final List<String> metadataFiles;
    final File backupDirectory;

    PatchDryRun(List<String> filesToAdd, List<String> filesToModify, List<String> metadataFiles, File backupDirectory) {
        this.filesToAdd = filesToAdd;
        this.filesToModify = filesToModify;
        this.metadataFiles = metadataFiles;
        this.backupDirectory = backupDirectory;
    }

    String describe() {
        StringBuilder builder = new StringBuilder();
        appendList(builder, "Files to add:", filesToAdd);
        builder.append('\n');
        appendList(builder, "Files to modify:", filesToModify);
        builder.append("\nBackup target:\n")
                .append(backupDirectory.getAbsolutePath())
                .append("\n\nApply Patch remains locked until backup/restore and Shizuku file writing are tested on a real device.");
        return builder.toString();
    }

    private static void appendList(StringBuilder builder, String title, List<String> values) {
        builder.append(title).append('\n');
        for (String value : values) {
            builder.append("- ").append(value).append('\n');
        }
    }
}

File: app/src/main/kotlin/com/acceleratorer/wuwavn/PatchDryRun.kt
package com.acceleratorer.wuwavn

import java.io.File

data class PatchDryRun(
    val filesToAdd: List<String>,
    val filesToModify: List<String>,
    val metadataFiles: List<String>,
    val backupDirectory: File,
) {
    fun describe(): String = buildString {
        appendList("Files to add:", filesToAdd)
        append('\n')
        appendList("Files to modify:", filesToModify)
        append("\nBackup target:\n")
        append(backupDirectory.absolutePath)
        append("\n\nUnlocked write flows are separate and verified: PAK-only install, Safe / Default config preset, and Remove Patch with MountLang rollback. Balanced is preview-only; Balanced write, Performance, and Max Graphics remain locked.")
    }

    private fun StringBuilder.appendList(title: String, values: List<String>) {
        append(title).append('\n')
        for (value in values) {
            append("- ").append(value).append('\n')
        }
    }
}
